### PR TITLE
Added support for `Editor.build` property

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Added support for `Editor.build` property. Closes #47.

---

### CODE REVIEW

File: src/editor/editor.js [hunk 1]
```diff
@@ -45,7 +47,7 @@ export default class Editor {
 		 * @readonly
 		 * @member {module:core/plugin~PluginCollection}
 		 */
-		this.plugins = new PluginCollection( this );
+		this.plugins = new PluginCollection( this, availablePlugins );
```
Comment (Reinmar) [inline src/editor/editor.js:50]: This is overly complicated. All we need here is `new PluginCollection( this, this.constructor.build && this.constructor.build.plugins )`

Or, at least remove the `findAvailablePlugins` function and define `availablePlugins` as:

```js
const availablePlugins = this.constructor.build && this.constructor.build.plugins;
```

**Comment by Reinmar** [comment]
This PR closes #67, not #47.

File: src/editor/editor.js [hunk 2]
```diff
@@ -189,6 +193,42 @@ export default class Editor {
 
 mix( Editor, EmitterMixin );
 
+// @param {module:utils/config~Config} config
+// @param {Object} builtInConfig
+// @param {Object.<String,*>} builtInConfig.config
+// @param {Array.<module:core/plugin~Plugin>} availablePlugins
+function extendEditorConfig( config, builtInConfig, availablePlugins ) {
```
Comment (Reinmar) [inline src/editor/editor.js:200]: This looks bad. It shouldn't require so much code in this class to set the defaults. It must be `Config`'s ability to easily define the defaults.

I also need to think what should extend what here. I'll let you know once I figure this out :D.

**Comment by pomek** [comment]
This PR is invalid (wrong branch name). I corrected it - https://github.com/ckeditor/ckeditor5-core/pull/72.

File: src/editor/editor.js (hunk 2 above)
Comment (pomek) [reply src/editor/editor.js:200]: Ok. When you will be ready, put your idea to https://github.com/ckeditor/ckeditor5-core/pull/72.

**Pull request closed without merging**